diff --git a/node_modules/dynamic-cdn-webpack-plugin/lib/index.js b/node_modules/dynamic-cdn-webpack-plugin/lib/index.js
index fefbdb4..99f3f0f 100644
--- a/node_modules/dynamic-cdn-webpack-plugin/lib/index.js
+++ b/node_modules/dynamic-cdn-webpack-plugin/lib/index.js
@@ -171,7 +171,9 @@ class DynamicCdnWebpackPlugin {
         includeAssetsPlugin.apply(compiler);
 
         compiler.hooks.afterCompile.tapAsync(pluginName, (compilation, cb) => {
-            const assets = Object.values(this.modulesFromCdn).map(moduleFromCdn => moduleFromCdn.url);
+            const values = Object.values(this.modulesFromCdn);
+            const sorted = values.sort((a, b) => a.weight > b.weight ? 1 : -1)
+            const assets = sorted.map(moduleFromCdn => moduleFromCdn.url);
 
             // HACK: Calling the constructor directly is not recomended
             //       But that's the only secure way to edit `assets` afterhand
